bpf/tests/lib/policy.h

```diff
@@ -1,19 +1,59 @@
 /* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
 /* Copyright Authors of Cilium */
 
+static __always_inline __u8
+policy_calc_wildcard_bits(__u8 protocol, __u16 dport)
+{
+	__u8 wildcard_bits = 0;
+
+	/* Wildcard the port: */
+	if (!dport) {
+		wildcard_bits += 16;
+
+		/* Only wildcard protocol if port is also wildcarded: */
+		if (!protocol)
+			wildcard_bits += 8;
+	}
+
+	return wildcard_bits;
+}
+
+static __always_inline void
+policy_delete_entry(bool egress, __u32 sec_label, __u8 protocol, __u16 dport)
+{
+	__u8 wildcard_bits = policy_calc_wildcard_bits(protocol, dport);
+	/* Start with an exact L3/L4 policy, and wildcard it as determined above: */
+	__u32 key_prefix_len = POLICY_FULL_PREFIX - wildcard_bits;
+
+	struct policy_key key = {
+		.lpm_key = { key_prefix_len, {} },
+		.sec_label = sec_label,
+		.egress = egress,
+		.protocol = protocol,
+		.dport = dport,
+	};
+
+	map_delete_elem(&cilium_policy_v2, &key);
+}
+
 static __always_inline void
-policy_add_entry(bool egress, __u32 sec_label, __u8 protocol, __u16 dport, bool deny,
-		 __u8 prefix_length)
```

Comment:
Maybe keep the prefix_length argument as `port_prefix_len` and automatically compute it like here if passed in as `0`? That way it would be possible to test partially wildcarded port policies :-)

Reply:
ack, I think I understand what you mean. But imo that warrants a distinct helper for port ranges - let's add it when we add coverage for those scenarios.